### PR TITLE
Adds skript list command

### PR DESCRIPTION
**Pull request by TheLimeGlass**
### Description
- Added `/skript list` command
- Added pattern option to the CommandHelp
- Cleanups

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/command/CommandHelp.java [hunk 1]
```diff
@@ -68,8 +74,27 @@ private CommandHelp(String command, String argsColor, String node) {
...
+	 * Add an argument to the CommandHelp.
+	 * 
+	 * @param argument String that represents the node path to the help info.
+	 * @param pattern The pattern to match against user input.
+	 * @return This CommandHelp for building.
+	 */
+	public CommandHelp add(String argument, @Nullable String pattern) {
 		ArgumentHolder holder = new ArgumentHolder(argument);
+		if (pattern != null)
+			holder.pattern = PatternCompiler.compile(pattern);
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/command/CommandHelp.java:97]: I don't see a useful reason for adding skript patterns to skript command arguments. A simple aliases param is enough.

File: src/main/java/ch/njol/skript/SkriptCommand.java [hunk 2]
```diff
@@ -446,6 +447,15 @@ else if (args[0].equalsIgnoreCase("test") && TestMode.DEV_MODE) {
 					);
 			}
 
+			else if (args[0].equalsIgnoreCase("list") || args[0].equalsIgnoreCase("show")) {
+				info(sender, "list.enabled", StringUtils.join(ScriptLoader.getLoadedScripts().stream()
+						.map(script -> script.getConfig().getFileName())
+						.toArray(String[]::new)));
+				info(sender, "list.disabled", StringUtils.join(ScriptLoader.getDisabledScripts().stream()
```
Comment (AyhamAl-Ali) [inline src/main/java/ch/njol/skript/SkriptCommand.java:454]: I think the message should include the path of the script as users can have scripts of the same name in different folders

File: src/main/resources/lang/english.lang [hunk 3]
```diff
@@ -109,6 +110,9 @@ skript command:
 			# invalid version: No changelog for the version <gold>%s<red> available
 			title: <bold><cyan>%s<reset> (%s)
 			next page: <grey>page %s of %s. Type <gold>/skript update changes %s<gray> for the next page (hint: use the up arrow key)
+	list:
+		enabled: <white>Enabled scripts: <lime green>%s
```
Comment (AyhamAl-Ali) [inline src/main/resources/lang/english.lang:114]: Can we have tooltips here? we can display the path of the script using tooltips

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]
Nice feature :zap:

File: src/main/java/ch/njol/skript/SkriptCommandTabCompleter.java [hunk 4]
```diff
@@ -116,6 +116,7 @@ public List<String> onTabComplete(CommandSender sender, Command command, String
 			options.add("enable");
 			options.add("disable");
 			options.add("update");
+			options.add("list");
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/SkriptCommandTabCompleter.java:119]: Might be worth including the `show` alias?

**Review by Moderocky — APPROVED** [review]

**Comment by sovdeeth** [comment]
closing due to inactivity

File: src/main/resources/lang/english.lang (hunk 3 above)
Comment (TheLimeGlass) [reply src/main/resources/lang/english.lang:114]: No it needs to go to console too

File: src/main/java/ch/njol/skript/SkriptCommand.java [hunk 5]
```diff
@@ -441,6 +444,22 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
...
+				ScriptLoader.getLoadedScripts().stream()
+						.map(script -> script.getConfig().getFileName())
+						.forEach(name -> info(sender, "list.enabled.element", name));
+				info(sender, "list.disabled.header");
+				ScriptLoader.getDisabledScripts().stream()
+						.flatMap(file -> {
+							if (file.isDirectory()) {
+								return Arrays.stream(file.listFiles());
+							}
+							return Arrays.stream(new File[]{file});
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/SkriptCommand.java:458]:
```suggestion
							return Stream.of(file);
```
might need import

**Review by APickledWalrus — APPROVED** [review]
seems ok

**Review by sovdeeth — APPROVED** [review]

**Pull request merged**